### ISSUE TITLE
48 tests failed out of 62 on Fedora 29

### ISSUE THREAD

**Issue by junghans** [espressopp/espressopp#265]
Details can be found here:
https://copr-be.cloud.fedoraproject.org/results/junghans/espressopp/fedora-29-x86_64/00861278-espressopp/build.log.gz

I set this up, so that you can edit 
https://github.com/espressopp/fedora-copr/blob/master/espressopp.spec
and it will automatically build here:
https://copr.fedorainfracloud.org/coprs/junghans/espressopp/builds/

**Comment by junghans** [comment]
On Fedora (even inside docker) one can reproduce that locally by:
```
mock -r fedora-29-x86_64 --init
wget https://raw.githubusercontent.com/espressopp/fedora-copr/master/espressopp.spec
spectool -g espressopp.spec
rpmbuild -D"_sourcedir ${PWD}" -D"_srcrpmdir ${PWD}" -bs espressopp.spec
mock -r fedora-29-x86_64 --no-clean espressopp-*.src.rpm
mock -r fedora-29-x86_64 --shell
```

**Comment by jkrajniak** [comment]
This looks familiar... maybe again somewhere the deletion of the element in the STL container is not done idiomatic.

**Comment by jkrajniak** [comment]
hm I try to run but I am getting following error
```
Complete!
Finish: dnf install
ERROR: Command failed: 
 # /usr/bin/systemd-nspawn -q -M ca9a4e9404eb4a378fa2f2b4b9a3ba3a -D /var/lib/mock/fedora-29-x86_64/root -a --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin --setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007" --setenv=PS1=<mock-chroot> \s-\v\$  --setenv=LANG=en_US.UTF-8 /usr/sbin/groupadd -g 135 mock

```

**Comment by junghans** [comment]
Hmm, I guess you need to run `systemd-machine-id-setup` before mock.

**Comment by junghans** [comment]
This seems to be https://github.com/rpm-software-management/mock/issues/96, adding the `--old-chroot` option seems to help.

**Comment by junghans** [comment]
Ok, I did:
```
docker run --cap-add=SYS_ADMIN -it fedora:29 /bin/bash
```
and then inside docker:
```
dnf install -y mock wget spectool
mock -r fedora-29-x86_64 --init
systemd-machine-id-setup
mock -r fedora-29-x86_64 --init --old-chroot
wget https://raw.githubusercontent.com/espressopp/fedora-copr/master/espressopp.spec
spectool -g espressopp.spec
rpmbuild -D"_sourcedir ${PWD}" -D"_srcrpmdir ${PWD}" -bs espressopp.spec
mock -r fedora-29-x86_64 --no-clean --old-chroot espressopp-*.src.rpm
```

**Comment by junghans** [comment]
Any update?

**Comment by jkrajniak** [comment]
Not really, however I could at last reproduce the error in docker but didn't have time to look further.

**Comment by junghans** [comment]
I would like to fix this and then make a v2.0.2 (as v2.0.1) never got deployed to dockerhub.

**Comment by jkrajniak** [comment]
Please, have a look on #271 ;
